lib/cursor.js

```diff
@@ -654,6 +669,10 @@ var nextFunction = function(self, callback) {
     self._find(function(err) {
       if (err) return handleCallback(callback, err, null);
 
+      if (self.cursorState.cursorId && self.cursorState.cursorId.isZero() && self._endSession) {
```

Comment:
where you running into cases where `self_endSession` didn't exist?

Reply:
Yes, this is a bigger issue I had to work around: [commandCursor selectively inherits methods from the base class](https://github.com/mongodb/node-mongodb-native/blob/3.0.0/lib/command_cursor.js#L128). I'm not sure why this is the case, and I wasn't sure if command cursors should have the same session management code (especially b/c when I tried to add it in, it led to >200 unit test failures :) ).